sockeye/inference.py

```diff
@@ -1515,22 +1484,23 @@ def _beam_search(self,
 
             # (6) Prune out low-probability hypotheses. Pruning works by setting entries `inactive`.
             if self.beam_prune > 0.0:
-                scores_accumulated, best_word_indices, inactive, finished = self._prune(scores_accumulated,
-                                                                                        best_word_indices,
-                                                                                        inactive,
-                                                                                        finished)
+                inactive = self.prune(scores_accumulated, finished)
+                best_word_indices = mx.nd.where(inactive, self.zeros_array, best_word_indices)
```

Comment:
I like the simplified signature for `prune()`. Since we set `best_word_indices` here after the call, it seems to me now that it would be clearer and simpler to do that here for `scores_accumulated`, instead of its reassignment in two separate places (below, for the same t, and above, at the next t).